balena/models/device.py

```diff
@@ -63,6 +64,58 @@ def __init__(self):
         self.release = Release()
         self.device_os = DeviceOs()
         self.hup = Hup()
+        
+    def __upsert_device_config_variable(self, device, name, value):
+        try:
+            data = {
+                'device': device['id'],
+                'name': name,
+                'value': value
+            }
+
+            return self.base_request.request(
+                'device_config_variable', 'POST', data=data,
+                endpoint=self.settings.get('pine_endpoint')
+            )
+        except exceptions.RequestError as e:
+            if 'Unique key constraint violated' in e.message or 'must be unique' in e.message:
```

Comment:
Here is how pineClient does this check in upsert.
See: https://github.com/balena-io-modules/pinejs-client-js/blob/4a4f8535ef5b3d824ebbc09b46589efecb3a73fe/src/index.ts#L1079-L1080

We can be sure that this will work since model v6, so the current major of the python SDK should probably be able to use it w/o issues.